__tests__/app.test.js

```diff
@@ -130,6 +130,45 @@ describe("GET /api/reviews/:review_id", () => {
   });
 });
 
+describe("POST /api/reviews/:review_id/comments", () => {
+  test("201, responds with the posted comment", () => {
+    const commentToAdd = { username: "mallionaire", body: "Great game!" };
+    return request(app)
+      .post("/api/reviews/4/comments")
+      .send(commentToAdd)
+      .expect(201)
+      .then((res) => {
+        const { comment } = res.body;
+        expect(comment).toHaveProperty("comment_id");
+        expect(comment).toHaveProperty("body", "Great game!");
+        expect(comment).toHaveProperty("votes", 0);
+        expect(comment).toHaveProperty("author", "mallionaire");
+        expect(comment).toHaveProperty("review_id", 4);
+        expect(comment).toHaveProperty("created_at");
+      });
+  });
+  test("400, responds with bad request when passed an incomplete comment input", () => {
+    const commentToAdd = { username: "mallionaire" };
+    return request(app)
+      .post("/api/reviews/4/comments")
+      .send(commentToAdd)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("Bad request - incomplete information");
+      });
+  });
+  test("400, responds with bad request when passed an incomplete comment input", () => {
+    const commentToAdd = { body: "Great game!" };
+    return request(app)
+      .post("/api/reviews/4/comments")
+      .send(commentToAdd)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("Bad request - incomplete information");
+      });
+  });
+});
+
```

Comment:
3x sad path test missing here:
404 - username does not exist
400 - invalid id "/api/reviews/not-a-number/comments"
404 - valid but non-existent id "/api/reviews/7777777/comments"